impeller/renderer/backend/vulkan/fence_waiter_vk.cc

```diff
@@ -54,56 +53,36 @@ void FenceWaiterVK::Main() {
     wait_set_cv_.wait(lock, [&]() { return !wait_set_.empty() || terminate_; });
 
     auto device_holder = device_holder_.lock();
-    if (!device_holder) {
+    if (!device_holder || terminate_) {
       break;
     }
 
-    auto wait_set = TrimAndCreateWaitSetLocked(device_holder);
-
+    std::vector<vk::UniqueFence> temp_wait_set = std::move(wait_set_);
+    std::vector<fml::closure> callbacks = std::move(wait_set_callbacks_);
     lock.unlock();
 
-    if (!wait_set.has_value()) {
-      break;
+    std::vector<vk::Fence> fences(temp_wait_set.size());
+    for (auto i = 0u; i < temp_wait_set.size(); i++) {
+      fences[i] = temp_wait_set[i].get();
     }
-    if (wait_set->empty()) {
+
+    if (fences.empty()) {
       continue;
     }
 
     auto result = device_holder->GetDevice().waitForFences(
-        wait_set->size(),                        // fences count
-        wait_set->data(),                        // fences
-        false,                                   // wait for all
+        fences.size(),                           // fences count
+        fences.data(),                           // fences
+        true,                                    // wait for all
         std::chrono::nanoseconds{100ms}.count()  // timeout (ns)
     );
     if (!(result == vk::Result::eSuccess || result == vk::Result::eTimeout)) {
```

Comment:
Previously if this was an error, the fence would be tried again later right?  Now it gets lost.

Reply:
nevermind, it would get lost in the old code too.

Reply:
it doesn't get lost, everything is dropped at the end of the scope here. The only other documented error condition is device loss which makes all of these invalid anyway.

Reply:
so barring the issue with timing out, I don't think there is a problem.